frontend/amundsen_application/static/js/components/Table/index.tsx

```diff
@@ -585,20 +588,31 @@ const TableRow: React.FC<TableRowProps> = ({
       onCollapse={onCollapse}
       rowValues={rowValues}
       onClick={setExpandedRows}
-      isSelectedRow={currentSelectedKey === columnKey}
       nestedLevel={nestedLevel}
     />
   );
+  const handleRowClick = () => {
+    onRowClick?.(rowValues, columnKey);
+  };
+  const frontendParsedNestedLevel = (rowValues.content as ContentType)
+    ?.nestedLevel;
+  const isFrontendParsedNestedColumn =
+    frontendParsedNestedLevel !== undefined && frontendParsedNestedLevel > 0;
 
   return (
     <React.Fragment key={columnKey}>
       <tr
         className={`ams-table-row ${
           rowValues.isNestedColumn ? 'is-nested-column-row' : ''
-        } ${currentSelectedKey === columnKey ? 'is-selected-row' : ''}`}
+        } ${currentSelectedKey === columnKey ? 'is-selected-row' : ''} ${
+          onRowClick && !isFrontendParsedNestedColumn
+            ? 'is-interactive-row'
+            : ''
+        }`}
```

Comment:
Nit: This is a bit too much now, I would create a 'rowClasses' temporal variable and move the logic there.